core/src/avm2/optimize.rs

```diff
@@ -769,17 +773,20 @@ pub fn optimize<'gc>(
                     }
 
                     if !stack_push_done {
-                        if let Some((class, index)) =
-                            outer_scope.get_entry_for_multiname(&multiname)
-                        {
-                            *op = Op::GetOuterScope {
-                                index: index as u32,
-                            };
+                        if let Some(info) = outer_scope.get_entry_for_multiname(&multiname) {
```

Comment:
can you add more comments here and in the function signature type?
Smth like "we hit a `with` scope, so we can't look deeper at all"

Alternatively, change `Option<Option<` to a 3-value enum (but I'm okay with just a comment here)